### PR TITLE
[ConsulOperations] Make the recurse query configurable

### PR DESCRIPTION
**Pull request by FrogDevelopper**
Make the read KV operation more flexible by allowing the recurse parameter to be set to `true` or `false`

### CODE REVIEW

File: discovery-client/src/main/java/io/micronaut/discovery/consul/client/v1/ConsulOperations.java [hunk 1]
```diff
@@ -48,33 +50,49 @@ public interface ConsulOperations {
...
      * @param key        The key
      * @param datacenter The data center
+     * @param recurse    Specifies if the lookup should be recursive and treat key as a prefix instead of a literal match.
      * @param raw        Whether the value should be raw without encoding or metadata
-     * @param seperator  The separator to use
+     * @param separator  The separator to use
      * @return A {@link Publisher} that emits a list of {@link KeyValue}
      */
-    @Get(uri = "/kv/{+key}?recurse=true{&dc}{&raw}{&seperator}", single = true)
+    @Get(uri = "/kv/{+key}?{&recurse}{&dc}{&raw}{&separator}", single = true)
```
Comment (FrogDevelopper) [inline discovery-client/src/main/java/io/micronaut/discovery/consul/client/v1/ConsulOperations.java:85]: also fixing a typo on the `separator` parameter name

**Comment by FrogDevelopper** [comment]
Hello
the build got this error
>Run graalvm/setup-graalvm@v1.2.1
  with:
    distribution: graalvm
    java-version: 17
    github-token: ***
    java-package: jdk
    set-java-home: true
    check-for-updates: true
    native-image-musl: false
    native-image-job-reports: false
    native-image-pr-reports: false
  env:
    GRADLE_ENTERPRISE_ACCESS_KEY: 
    GRADLE_ENTERPRISE_CACHE_USERNAME: 
    GRADLE_ENTERPRISE_CACHE_PASSWORD: 
    GH_TOKEN_PUBLIC_REPOS_READONLY: 
    GH_USERNAME: 
    TESTCONTAINERS_RYUK_DISABLED: true
    PREDICTIVE_TEST_SELECTION: true
    SONAR_TOKEN: 
    GITHUB_TOKEN: ***
Error: Failed to download graalvm-jdk-17_linux-x6[4](https://github.com/micronaut-projects/micronaut-discovery-client/actions/runs/11826248768/job/32951796684#step:5:4)_bin.tar.gz. Are you sure java-version: '17' is correct?

How can I fix it ? it is on the github action workflow, so not related to this PR's changes

**Comment by radovanradic** [comment]
> Hello the build got this error
> 
> > Run graalvm/setup-graalvm@v1.2.1
> > ...
> > Error: Failed to download graalvm-jdk-17_linux-x6[4](https://github.com/micronaut-projects/micronaut-discovery-client/actions/runs/11826248768/job/32951796684#step:5:4)_bin.tar.gz. Are you sure java-version: '17' is correct?
> 
> How can I fix it ? it is on the github action workflow, so not related to this PR's changes

Try changing line https://github.com/FrogDevelopment/micronaut-discovery-client/blob/5.0.x/.github/workflows/gradle.yml#L48 from `uses: graalvm/setup-graalvm@v1.2.1` to `uses: graalvm/setup-graalvm@v1.2.5`

**Comment by FrogDevelopper** [comment]
I'm adding some changes to it, so I'll close for now the PR

**Pull request closed without merging**